### ISSUE TITLE
Doesn't build

### ISSUE THREAD

**Issue by limitedAtonement** [david-bergstrom/generals-bot#1]
After building the project, I ran it and got the following response:

lmat@ryle~/source/generals-bot$ ./main
Tries to make a connection to: http://botws.generals.io
Error: No active session
[2017-02-12 09:13:09] [connect] Successful connection
[2017-02-12 09:13:09] [error] Server handshake response error: websocketpp.processor:21 (A required HTTP header is missing)
^C
lmat@ryle~/source/generals-bot$

**Comment by limitedAtonement** [comment]
I verified that commit fixes the problem, but I have no idea how! How did you decide to move the declaration (and construction) into main? The output is now:

    Tries to make a connection to: http://botws.generals.io
    Error: No active session
    [2017-02-12 14:12:03] [connect] Successful connection
    [2017-02-12 14:12:03] [connect] WebSocket Connection 45.55.8.225:80 v-2 "WebSocket++/0.7.0" /socket.io/?EIO=4&transport=websocket&t=1486926723 101
    Connected
    ^C
    lmat@ryle~/source/generals-bot$
`

**Comment by david-bergstrom** [comment]
Great to hear it works for you as well! Maybe I closed the issue a bit prematurely, without checking with you first.

I don't know why it doesn't work the other way around. I suspect it is an issue with the socket.io-client-cpp library or maybe some of its underlying libraries. I ran git-bisect and found that the error was introduced in faa8f0861b679ed88ab75918cc1e723dd639c3a3. Going through the changes made their I realised the only difference (other than abstraction) was that the declaration of the sio_client was made outside of main.
